TUTORIAL.md

```diff
@@ -0,0 +1,371 @@
+
+Upload this datafile from your local machine to your newly created aws bucket
+
+```
+order_id,customer_id,purchase_date,amount
+ORDER1,CUST1,1/1/2021,100
+ORDER2,CUST2,2/2/2022,200
+ORDER3,CUST3,3/3/2023,300
+```
+
+### Set up Snowflake
+
+Sign up for a [free Snowflake trial](https://signup.snowflake.com/), selecting the Enterprise Edition.
+
+Once your workspace is ready, navigate to a worksheet and enter these commands:
+
+```sql
+create warehouse ASTRO_SDK_DW;
+create database ASTRO_SDK_DB;
+create schema ASTRO_SDK_SCHEMA;
+```
+Then, in the classic Snowflake console in the upper right, select the role, warehouse, database and schema
+to look like this:
+
+![classic-console](docs/images/snowflake-classic-console.png)
+
+Note that you can set the warehouse, database, and schema names to something else, but you'll just
+need to be consistent with whatever names you choose throughout the remainder of this tutorial.
+
+## Install Astro-SDK on your local machine
+
+* Run the following command to install the Python SDK.
+
+  ```shell
+  pip install 'astro-sdk-python>=0.9'
+  ```
+
+  You can also install dependencies for specific providers by specifying "extras",
+  example, the following command install all the dependencies for Amazon & Snowflake:
+
+  ```shell
+  pip install 'astro-sdk-python[amazon,snowflake]>=0.9'
+  ```
+
+  Or you can also those providers separately using the following command. (We recommend using "extras")
+
+  ```shell
+  pip install apache-airflow-providers-amazon
+  pip install apache-airflow-providers-snowflake
+  ```
+
+* Create the following environment variables
+
+  ```shell
+  export AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True
+  export AIRFLOW__ASTRO__SQL_SCHEMA=<snowflake_schema>
+  ```
+
+If you're using macOS, set this environment variable too [(background)](https://github.com/apache/airflow/issues/12808):
+
+  ```shell
+  export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
+  ```
+
+* Start Airflow by running `airflow standalone` command and check out the Webserver
+at [`http://localhost:8080/`](http://localhost:8080/).
+
+## Setup Airflow connections
+
+* In your Astronomer Airflow UI select Admin-->Connectors
+
+![connections](docs/images/airflow-webserver-connection.png)
+
+
+### Connect S3 to Airflow
+
+Click on the blue "+" icon to *Add a new record*
+
+Set these fields:
+
+* Connection Id: `aws_default`
+* Connection Type: `S3`
+* Extra: `{"aws_access_key_id": "<your_access_key>", "aws_secret_access_key": "<you_secret_access_key>"}`
+
+### Connect Snowflake to Airflow
+
+Click on the blue "+" icon to *Add a new record*
+
+* Connection Id: `snowflake_default`
+* Connection Type: `Snowflake`
+* Host: `https://<account>.<region>.snowflakecomputing.com/`. This is the URL where you can log into your Snowflake account
+* Schema:
+* Login:
+* Password:
+* Account:
+* Database:
+* Region: (something like us-east-1 or us-central1.gcp)
+* Role:
+* Warehouse:
```

Comment:
If we set this information as part of the connection, the table declarations will use them - simplifying the DAG